src/nrncvode/cvodeobj.cpp

```diff
@@ -1377,29 +1399,50 @@ void Cvode::statistics() {
 
 void Cvode::matmeth() {
 	switch ( ncv_->jacobian() ) {
-	case 1:
-		CVDense(mem_, neq_);
-		break;
+    case 1:
+    {
+        /* Create dense SUNMatrix for use in linear solver */
+        SUNMatrix A = SUNDenseMatrix(neq_, neq_);
+        assert(A);
+
+        /* Create dense SUNLinearSolver object for use by CVode */
+        SUNLinearSolver LS = SUNDenseLinearSolver(y_, A);
+        assert(LS);
+
+        /* attach the matrix and linear solver to CVode */
+        int flag = CVDlsSetLinearSolver(mem_, LS, A);
+        assert(flag);
+
+        /* Set jacobian and allocate memory */
+        /* (TODO Michael Hines: is there a jacobian for dense matrix
+         * in neuron? If yes, add it here instead of NULL */
+        flag = CVDlsSetJacFn(mem_, NULL);
+        if (flag == CVDLS_MEM_FAIL)
+          throw std::runtime_error(
+              "ERROR: can't allocate memory for dense jacobian");
+        break;
+    }
 	case 2:
 		CVDiag(mem_);
 		break;
 	default:
-		((CVodeMem)mem_)->cv_linit = minit;
+    {
+        /* CVODES guide chapter 8: Providing Alternate Linear
+         * Solver Modules: only lsolve function is mandatory
+         * (non-used functions need to be set to null) */
+        ((CVodeMem)mem_)->cv_linit = nullptr;
```

Comment:
for c++11 related code (`nullptr`), we have to later add checks in `configure` to make sure : 
- compiler support c++11
- c++ flags are added for the same